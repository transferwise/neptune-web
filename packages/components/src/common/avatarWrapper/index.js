import AvatarWrapper from './AvatarWrapper';

export default AvatarWrapper;
