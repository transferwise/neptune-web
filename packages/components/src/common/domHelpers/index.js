export function isTouchDevice() {
  return !!window?.matchMedia('(pointer: coarse)')?.matches;
}

export {
  addClickClassToDocumentOnIos,
  removeClickClassFromDocumentOnIos,
} from './documentIosClick';
