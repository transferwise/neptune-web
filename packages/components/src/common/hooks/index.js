export { default as useHasIntersected } from './useHasIntersected';
export { default as useClientWidth } from './useClientWidth';
