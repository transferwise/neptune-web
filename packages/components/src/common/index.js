export { default as HistoryNavigator } from './historyNavigator';
export { addNoScrollBodyClass, removeNoScrollBodyClass } from './DOMOperations';
export { Theme } from './theme';
// Props
export { Breakpoint } from './propsValues/breakpoint';
export { Size } from './propsValues/size';
export { Type } from './propsValues/type';
export { DateMode } from './propsValues/dateMode';
export { MonthFormat } from './propsValues/monthFormat';
// Position and Placement are duplicated and needs to be aligned with a breaking change
export { Position } from './propsValues/position';
export { Layout } from './propsValues/layouts';
export { FormControlType } from './propsValues/formControlType';
export { Status } from './propsValues/status';
export { Sentiment } from './propsValues/sentiment';
export { ProfileType } from './propsValues/profileType';
export { Key } from './key';
