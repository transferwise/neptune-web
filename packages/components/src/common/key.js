export const Key = {
  ENTER: 'Enter',
  SPACE: [' ', 'Spacebar'],
};
