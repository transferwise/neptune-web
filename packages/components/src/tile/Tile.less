@import (reference) '@transferwise/neptune-css/src/less/variables/_typography.less';

.np-tile {
  min-width: 120px;
  border-radius: var(--radius-small);
  width: 100%;

  &.decision {
    margin-top: 0;
    border: 1px solid var(--color-border-neutral);
  }

  &__media {
    width: 100%;
  }

  &__title {
    margin-top: var(--size-16);
    font-size: 1.333333rem; // 20px
    line-height: calc(24 / 20); // 24px
    color: var(--color-text-primary);
  }

  &__description {
    margin-top: var(--size-12);
    line-height: calc(22 / 16); // 22px
    color: var(--color-text-secondary);
    font-weight: @font-weight-base;
  }

  &--small {
    .np-tile__title {
      margin-top: var(--size-12);
      font-size: @font-size-base; //16px
      line-height: calc(22 / 16); // 22px
    }

    .np-tile__description {
      margin-top: var(--size-4);
    }
  }

  &:hover {
    .tw-avatar {
      pointer-events: none;

      .tw-avatar__content {
        background-color: var(--color-background-screen);
      }
    }
  }
}
