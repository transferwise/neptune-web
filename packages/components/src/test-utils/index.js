import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from '..';
import { DEFAULT_LOCALE } from '../common/locale';
import en from '../../i18n/en.json';

/**
 * Custom `render` function which wraps passed elements in Provider component
 * Source: https://testing-library.com/docs/example-react-intl/#creating-a-custom-render-function
 */
function customRender(ui, { locale = DEFAULT_LOCALE, messages = en, ...renderOptions } = {}) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return <Provider i18n={{ locale, messages }}>{children}</Provider>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { userEvent };
export { customRender as render };
