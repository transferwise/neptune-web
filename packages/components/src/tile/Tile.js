import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';

import { Size, Key } from '../common';

import './Tile.css';

export const Tile = ({ className, description, disabled, href, media, onClick, size, title }) => {
  const isSmall = size === Tile.Size.SMALL;

  return (
    <a
      className={classNames(
        'decision',
        'flex-column',
        'np-tile',
        'text-no-decoration',
        'text-xs-center',
        className,
        {
          'p-y-5 p-x-4': !isSmall,
          'np-tile--small p-a-2': isSmall,
          disabled,
        },
      )}
      href={href}
      onClick={disabled ? null : onClick}
      onKeyDown={
        disabled
          ? null
          : ({ key }) => {
              if (key === Key.ENTER || Key.SPACE.indexOf(key) > -1) {
                onClick();
              }
            }
      }
      aria-label={title}
    >
      <div className="np-tile__media d-flex justify-content-center">{media}</div>
      <div className="np-tile__title">{title}</div>
      {description && <div className="np-tile__description">{description}</div>}
    </a>
  );
};

Tile.Size = Size;

Tile.propTypes = {
  /** Classes to apply to the Tile container */
  className: Types.string,
  description: Types.node,
  disabled: Types.bool,
  href: Types.string.isRequired,
  /** Accepts only Avatar and images */
  media: Types.node.isRequired,
  /** Function called onClick or onKeyDown */
  onClick: Types.func.isRequired,
  /** The size applied to Tile */
  size: Types.oneOf([Tile.Size.SMALL, Tile.Size.MEDIUM]),
  title: Types.node.isRequired,
};

Tile.defaultProps = {
  className: '',
  description: null,
  disabled: false,
  size: Tile.Size.MEDIUM,
};

export default Tile;
