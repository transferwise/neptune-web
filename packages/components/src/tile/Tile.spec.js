import React from 'react';
import { render, fireEvent } from '../test-utils';

import { Key } from '../common/key';

import Tile from '.';

describe(Tile, () => {
  const onClick = jest.fn();
  const defaultProps = {
    description: 'Receive money from friends and family like a local',
    href: '#href',
    media: <svg data-testid="illustration" />,
    onClick: () => onClick(),
    title: 'Receive money',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot when regular size', () => {
    const { asFragment } = render(<Tile {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when small size', () => {
    const { asFragment } = render(<Tile size={Tile.Size.SMALL} {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onClick when anchor is clicked', () => {
    const { getByText } = render(<Tile {...defaultProps} />);

    const title = getByText('Receive money');

    expect(onClick).not.toHaveBeenCalled();

    fireEvent.click(title);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('calls onClick when space or enter keys are pressed', () => {
    const { getByText } = render(<Tile {...defaultProps} />);

    const title = getByText('Receive money');

    fireEvent.keyDown(title, { key: Key.ENTER });

    expect(onClick).toHaveBeenCalledTimes(1);

    fireEvent.keyDown(title, { key: Key.SPACE[0] });

    expect(onClick).toHaveBeenCalledTimes(2);
  });

  it(`doesn't call onClick when disabled anchor is clicked`, () => {
    const { getByText } = render(<Tile {...defaultProps} disabled />);

    fireEvent.click(getByText('Receive money'));

    expect(onClick).not.toHaveBeenCalled();
  });

  it(`doesn't call onClick when space or enter keys are pressed on disabled anchor`, () => {
    const { getByText } = render(<Tile {...defaultProps} disabled />);

    const title = getByText('Receive money');

    fireEvent.keyDown(title, { key: Key.ENTER });

    expect(onClick).not.toHaveBeenCalled();

    fireEvent.keyDown(title, { key: Key.SPACE[0] });

    expect(onClick).not.toHaveBeenCalled();
  });
});
