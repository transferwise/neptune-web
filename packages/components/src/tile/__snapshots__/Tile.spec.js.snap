// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile matches snapshot when regular size 1`] = `
<DocumentFragment>
  <a
    aria-label="Receive money"
    class="decision flex-column np-tile text-no-decoration text-xs-center p-y-5 p-x-4"
    href="#href"
  >
    <div
      class="np-tile__media d-flex justify-content-center"
    >
      <svg
        data-testid="illustration"
      />
    </div>
    <div
      class="np-tile__title"
    >
      Receive money
    </div>
    <div
      class="np-tile__description"
    >
      Receive money from friends and family like a local
    </div>
  </a>
</DocumentFragment>
`;

exports[`Tile matches snapshot when small size 1`] = `
<DocumentFragment>
  <a
    aria-label="Receive money"
    class="decision flex-column np-tile text-no-decoration text-xs-center np-tile--small p-a-2"
    href="#href"
  >
    <div
      class="np-tile__media d-flex justify-content-center"
    >
      <svg
        data-testid="illustration"
      />
    </div>
    <div
      class="np-tile__title"
    >
      Receive money
    </div>
    <div
      class="np-tile__description"
    >
      Receive money from friends and family like a local
    </div>
  </a>
</DocumentFragment>
`;
